mam/v2/sponge/spongos_types.c

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * MAM is based on an original implementation & specification by apmi.bsu.by
+ * [ITSec Lab]
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "mam/v2/sponge/spongos_types.h"
+#include "mam/v2/pb3/pb3.h"
+
+size_t mam_spongos_serialized_size(mam_spongos_t const *const spongos) {
+  return pb3_sizeof_size_t(spongos->pos) + MAM2_SPONGE_WIDTH;
+}
+
+void mam_spongos_serialize(mam_spongos_t const *const spongos, trits_t trits) {
+  pb3_encode_size_t(spongos->pos, &trits);
+  pb3_encode_ntrytes(trits_from_rep(MAM2_SPONGE_WIDTH, spongos->sponge->state),
```

Comment:
Maybe we should have sponge serialization operations and call them in the spongos 🤔

Reply:
We could, but i prefer not, for 2 reason
1.We wouldn't need to serialize sponge by itself
2.We will have to separate sponge to serialize it because of circular dependency with pb3